Mathlib/SetTheory/Ordinal/Nimber.lean

```diff
@@ -357,4 +361,210 @@ theorem add_trichotomy {a b c : Nimber} (h : a + b + c ≠ 0) :
   · rw [← hx'] at hx
     exact Or.inr <| Or.inr hx
 
+theorem two_nsmul (a : Nimber) : 2 • a = 0 := by
+  rw [succ_nsmul, one_nsmul, add_self]
+
+theorem two_zsmul (a : Nimber) : (2 : ℤ) • a = 0 :=
+  two_nsmul a
+
+theorem add_eq_iff_eq_add : a + b = c ↔ a = c + b :=
+  sub_eq_iff_eq_add
+
+theorem eq_add_iff_add_eq : a = b + c ↔ a + c = b :=
+  eq_sub_iff_add_eq
+
+/-! ### Nimber multiplication -/
+
+/-- Nimber multiplication is recursively defined so that `a * b` is the smallest nimber not equal to
+`a' * b + a * b' + a' * b'` for `a' < a` and `b' < b`. -/
+-- We write the binders like this so that the termination checker works.
+protected def mul (a b : Nimber.{u}) : Nimber.{u} :=
+  sInf {x | ∃ a', ∃ (_ : a' < a), ∃ b', ∃ (_ : b' < b),
+    Nimber.mul a' b + Nimber.mul a b' + Nimber.mul a' b' = x}ᶜ
+termination_by (a, b)
```

Comment:
This is the first time I've seen `termination_by` used with a set literal!

Reply:
I also do this with the definition of nimber addition! Very useful tactic.